src/main/java/mate/academy/repository/BookRepository.java

```diff
@@ -1,8 +1,26 @@
 package mate.academy.repository;
 
+import java.util.List;
+import java.util.Optional;
 import mate.academy.model.Book;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.jpa.domain.Specification;
+import org.springframework.data.jpa.repository.EntityGraph;
 import org.springframework.data.jpa.repository.JpaRepository;
 import org.springframework.data.jpa.repository.JpaSpecificationExecutor;
+import org.springframework.stereotype.Repository;
 
+@Repository
 public interface BookRepository extends JpaRepository<Book, Long>, JpaSpecificationExecutor<Book> {
+    @EntityGraph(attributePaths = "categories")
+    Optional<Book> findById(Long id);
+
+    @EntityGraph(attributePaths = "categories")
+    Page<Book> findAll(Specification<Book> spec, Pageable pageable);
+
+    @EntityGraph(attributePaths = "categories")
+    Page<Book> findAll(Pageable pageable);
+
+    List<Book> findByCategories_Id(Long categoryId);
```

Comment:
does it work?

I though it should be `findByCategory_Id`

Reply:
it works pretty well.